### ISSUE TITLE
'default' filter in strict mode.

### ISSUE THREAD

**Issue by thomashunziker** [mbosecke/pebble#138]
The filter 'default' seems not to work as expected in strict mode. Test case:

```
   /**
     * Tests if the {@link DefaultFilter} is working as expected.
     *
     * @throws Exception
     *             thrown when something went wrong.
     */
    @Test
    public void testDefaultFilter() throws Exception {

        Loader<?> loader = new StringLoader();
        PebbleEngine pebble = new PebbleEngine(loader);
        pebble.setStrictVariables(true);

        PebbleTemplate template = pebble.getTemplate("{{ name | default('test') }}");

        Map<String, Object> context = new HashMap<>();

        Writer writer = new StringWriter();
        template.evaluate(writer, context);
        assertEquals("test", writer.toString());

    }
```
This test case throws: 

```
com.mitchellbosecke.pebble.error.RootAttributeNotFoundException: Root attribute [name] does not exist or can not be accessed and strict variables is set to true. ({{ name | default('test') }}:1)
	at com.mitchellbosecke.pebble.node.expression.ContextVariableExpression.evaluate(ContextVariableExpression.java:44)
	at com.mitchellbosecke.pebble.node.expression.FilterExpression.evaluate(FilterExpression.java:54)
	at com.mitchellbosecke.pebble.node.expression.FilterExpression.evaluate(FilterExpression.java:54)
	at com.mitchellbosecke.pebble.node.PrintNode.render(PrintNode.java:32)
	at com.mitchellbosecke.pebble.node.BodyNode.render(BodyNode.java:46)
	at com.mitchellbosecke.pebble.node.RootNode.render(RootNode.java:32)
	at com.mitchellbosecke.pebble.template.PebbleTemplateImpl.buildContent(PebbleTemplateImpl.java:62)
	at com.mitchellbosecke.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:112)
	at com.mitchellbosecke.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:83)
	at com.mitchellbosecke.pebble.StrictModeTest.testDefaultFilter(StrictModeTest.java:98)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)

```

The reason is clear: We check if the variable exists before we pipe it to the default filter. Hence we throw the exception before we even reach the default filter.

Has anyone a good idea to solve it?

**Comment by mbosecke** [comment]
I just committed a fix for this, a7e27ef. I actually used the same type of fix you introduced in #119.

**Comment by thomashunziker** [comment]
Cool. Thanks!
